Doc/library/zipfile.rst

```diff
@@ -374,13 +405,23 @@ ZipFile Objects
    .. versionchanged:: 3.6.2
       The *path* parameter accepts a :term:`path-like object`.
 
+   .. versionchanged:: next
+      The *preserve_permissions* parameter was added.
+
 
-.. method:: ZipFile.extractall(path=None, members=None, pwd=None)
+.. method:: ZipFile.extractall(path=None, members=None, pwd=None, \
+                               preserve_permissions=PreserveMode.NONE)
 
    Extract all members from the archive to the current working directory.  *path*
    specifies a different directory to extract to.  *members* is optional and must
    be a subset of the list returned by :meth:`namelist`.  *pwd* is the password
-   used for encrypted files as a :class:`bytes` object.
+   used for encrypted files as a :class:`bytes` object.  *preserve_permissions*
+   controls how the permissions of zipped files are preserved. The default is
+   :data:`PreserveMode.NONE` which does not preserve any permissions.  Other options are
+   to preserve a safe subset of permissions (:data:`PreserveMode.SAFE`) or all
+   permissions (:data:`PreserveMode.ALL`). If the archive was created on Windows,
+   the *preserve_pemissions* argument is ignored and permissions are not preserved.
```

Comment:
This documentation appears to be a duplication. That's not great. Probably better is to document the behavior in one place and reference that documentation. Keeping these two paragraphs in sync is going to be increasingly difficult with this addition. Maybe something like:

```
   Extract all members from the archive to the current working directory. *members* is optional and must
   be a subset of the list returned by :meth:`namelist`.

   *path*, *pwd*, and *preserve_permissions* have the same meaning as for :meth:`extract`.
```